src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -845,6 +844,7 @@ public NamedPlace getToPlace() {
     public RoutingRequest clone() {
         try {
             RoutingRequest clone = (RoutingRequest) super.clone();
+            clone.modes = modes.clone();
```

Comment:
Rather than just adding non-transit itinerary support, this line appears to be correcting a bug where modes were not replicated. Is that correct?

Reply:
Yes.